.github/workflows/gen_debian11.yml

```diff
@@ -0,0 +1,92 @@
+
+name: debian11
+
+on:
+  pull_request:
+    branches:
+    - main
+    paths-ignore:
+    - '.cirrus.yml'
+    - 'docs/*'
+    - 'ci/build-docs.sh'
+    - 'ci/generate-docs.py'
+    - 'ci/subst-release-info.py'
+    - '.github/workflows/pages.yml'
+    - '.github/ISSUE_TEMPLATE/*'
+    - '**/*.md'
+
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+    runs-on: "ubuntu-latest"
+    container: "debian:11"
+    steps:
+    - name: "set APT to non-interactive"
+      shell: bash
+      run: "echo 'debconf debconf/frontend select Noninteractive' | debconf-set-selections"
+    - name: "Update APT"
+      shell: bash
+      run: "apt update"
+    - name: "Install git"
+      shell: bash
+      run: "apt-get install -y git"
+    - name: "Install curl"
+      shell: bash
+      run: "apt-get install -y curl"
+    - name: "checkout repo"
+      uses: actions/checkout@v2
+      with:
+         submodules: "recursive"
+    - name: "Fetch tags"
+      shell: bash
+      run: "git fetch --depth=1 origin +refs/tags/*:refs/tags/*"
+    - name: "Fetch tag/branch history"
+      shell: bash
+      run: "git fetch --prune --unshallow"
+    - name: "Install Rust"
+      uses: actions-rs/toolchain@v1
+      with:
+         profile: "minimal"
+         toolchain: "stable"
+         override: true
+         components: "rustfmt"
+      env:
+         ACTIONS_ALLOW_UNSECURE_COMMANDS: "true"
+    - name: "Cache cargo"
+      uses: actions/cache@v2.1.6
+      with:
+         path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+            target
+
+         key: "debian11-None-2-${{ runner.os }}-${{ hashFiles('**/Cargo.lock') }}-cargo"
+    - name: "Install System Deps"
+      shell: bash
+      run: " env PATH=$PATH ./get-deps"
+    - name: "Check formatting"
+      shell: bash
+      run: "cargo fmt --all -- --check"
```

Comment:
Isn't it wasteful to do this for every OS? Formatting should be the same for all, right?

Reply:
Formatting doesn't cost anything compared to the rest of the stuff that CI does, so I'm not worried about this